2021-01-Winter/0_modified_binary_search/README.md

```diff
@@ -99,57 +141,207 @@ Go to [Solution](#s3)   [Top](#top)
 
 ### 1. SOLUTION 1 TODO :bug:
 
-Source: TODO :bug:
-
-#### Naive/Simple Solution
-
-TODO :bug:
-
-#### Optimal Solution
-
-TODO :bug:
-
-#### Testing The Solutions OR Driver For Solution
-
-TODO :bug:
+Source: https://leetcode.com/problems/binary-search/
+
+#### Binary Search Solution
+
+1. First, lets initiate our `left` and `right` variables, our left will start at the 0th index of the array, and right will start at `arrays.length -1` (the very last index of the array.)
```

Comment:
Nice! This looks like it was a lot of work! 
In my opinion, you don't need to go into this level of detail and write out the code :)  We just need a high level abstract explanation of your algorithm. 

The comments in your solution are actually a great example of that.

One thing that might be nice to include is a small example showing step-by-step what would happen at each step fo the algorithm to show that it works. 
This is also very similar to what you're expected to do in interviews, so it's great practice.